lib/librdkafka/NProducer.js

```diff
@@ -272,7 +259,16 @@ class NProducer extends EventEmitter {
       throw new Error("partitionCount must be number.");
     }
 
-    return this._murmur(key) % partitionCount;
+    switch (this._murmurHashVersion) {
+    case "2":
+      return murmur2Partitioner.partition(key, partitionCount);
+        
+    case "3":
+      return murmur.v3(key) % partitionCount;
+        
+    default:
+      throw new Error(`${this._murmurHashVersion} is not a supported murmur hash version. Choose '2' or '3'.`);
```

Comment:
this throws on every message you are trying to produce, evaluating this in the constructor only throws once and kills the producer. I personally think that would be a better dev-experience.